### DISCUSSION TITLE
Right configuration in energy panel

### DISCUSSION THREAD

**Discussion by matteoa66** [hultenvp/solis-sensor#245] [General]
Hi to all, I start this discussion to share and understand the right config in energy panel. (feedback welcome!!)
I'm using:

Grid consumption:
Daily Grid Energy Purchased

return to grid:
Daily On-grid Energy

Solar panel:
Energy Today

Battery system:
Daily Energy Discharged
Daily Energy Charged

@hultenvp if this info are right, can I suggest you to add in the readme?
Ciao, Matteo

**Comment by svetj** [comment]
hello. I think this is a good question.
I used the same parameters.
Someone can help us?
Thank you

**Reply by wyliecoyoteuk** [reply]
A list of the entities and their actual meaning might help people setting up.
I am struggling to match the entities with the correct values as shown in the Soliscloud web pages.
(not helped by the fact that the Soliscloud app seems to update at different times to the API)
For example:
Which of these is the exported energy figure?
On-grid energy
Grid energy purchased
Grid energy used
Power grid total power

Does total consumption mean total from all sources (battery, pv, grid) or just PV?
What does total energy used refer to if not total consumption?
etc., etc.

**Reply by wyliecoyoteuk** [reply]
Well spent some time on this, here are my guesses

Solis AC Output Total Power = current total AC output from panels
Solis Battery Power = live power input/output from batteries
solis daily energy purchased: total for the day of kWh grid energy input
solis daily energy used: total for the day of kWh energy used
Solis Total Energy Used: running total of all energy used
solis daily on-grid energy: total for the day of kWh energy exported
Solis Total Energy Purchased:running total of all grid energy input 
Solis Total On-grid Energy:running total of all grid energy exported
Solis Energy Today: running total of solar output
Solis AC Output Total Power: live solar AC output
Solis Power Grid total power: live in/out power 
Solis Total Consumption power: live power consumption from all sources, (includes discharging battery or charging from solar)

I notice that  "daily energy purchased" shows as less than "daily energy used", the difference looks to be the same as the amount of solar production plus the export amount.
Today I only used grid power in a 4 hour cheap rate window, and "Consumption Power" seems to be energy used **excluding** battery charging from the grid, but **including** battery charging from solar, whereas "Energy purchased" is all grid input including battery charging from the grid.
![image](https://user-images.githubusercontent.com/79256941/216641144-e872c410-5d04-45a7-83d8-8464fc74b5b3.png)

**Reply by hultenvp** [reply]
I like this discussion. Perhaps we can create a consolidated overview on the [wiki](https://github.com/hultenvp/solis-sensor/wiki) with pro's and con's on sensors and their indeed confusing semantics.

**Reply by wyliecoyoteuk** [reply]
That would be good. It doesn't help that the Soliscloud web pages use different terms as well.

**Reply by svetj** [reply]
Another thing I'd like to have on dashboard is the remaining battery:
I can see that going to Settings->device and services -> device
The last column shows the remainig battery for every device in %, and this is the same shown on Solis cloud.

I'd like also to have the remaing value in KW/h

Is it possibile?

PS. the wiki would be really great!

**Reply by TheVeryAngryUnicorn** [reply]
sensor.home_remaining_battery_capacity shows the state of charge

**Comment by TheVeryAngryUnicorn** [comment]
i would also really appreciate a wiki page detailing what to put where on the energy page, and any useful cards people have built.

presently i think i need to add solar production of "Home AC Output Power" as the solar panels, as this is reading the correct watts, and shares the correct symbol, but does not show up as an option under "add solar production"

confused :D

**Reply by wyliecoyoteuk** [reply]
I used "Solis Energy Total" (sensor.solis_total_energy) as the live energy panel output.

**Reply by viking2010** [reply]
> i would also really appreciate a wiki page detailing what to put where on the energy page, and any useful cards people have built.
> 
> presently i think i need to add solar production of "Home AC Output Power" as the solar panels, as this is reading the correct watts, and shares the correct symbol, but does not show up as an option under "add solar production"
> 
> confused :D

I've listed below the names I've given the sensors on my HA dashboard, along with the sensor names from the integration. I called my inverter "Solis Inverter", but you may have a different name after the sensor. prefix.

Hope this helps!

- Current PV Power: sensor.solis_inverter_ac_output_total_power
- PV1 Array Power: sensor.solis_inverter_dc_power_pv1
- PV2 Array Power: sensor.solis_inverter_dc_power_pv2
- House Load: sensor.solis_inverter_total_consumption_power
- Power To (+) & From (-) The Grid: sensor.solis_inverter_power_grid_total_power
- Last Status Update: sensor.solis_stats_received
- Consumption Today: sensor.solis_inverter_daily_grid_energy_used
- Inverter Temperature: sensor.solis_inverter_temperature
- Daily Generation: sensor.solis_inverter_energy_today
- Monthly Generation: sensor.solis_inverter_energy_this_month
- Total Generation: sensor.solis_inverter_energy_total
- Energy From The Grid (Today): sensor.solis_inverter_daily_grid_energy_purchased
- Energy to The Grid (Today): sensor.solis_inverter_daily_on_grid_energy
- Energy From The Grid (Total): sensor.solis_inverter_total_energy_purchased
- Energy to The Grid (Total): sensor.solis_inverter_total_on_grid_energy
- Battery Charge: sensor.solis_inverter_remaining_battery_capacity
- Battery SOH: sensor.solis_inverter_battery_state_of_health
- Battery Power: sensor.solis_inverter_battery_power
- Battery Voltage: sensor.solis_inverter_battery_voltage
- Battery Current: sensor.solis_inverter_battery_current
- Energy To The Battery (Today): sensor.solis_inverter_daily_energy_charged
- Energy From The Battery (Today): sensor.solis_inverter_daily_energy_discharged
- Energy To The Battery (Total): sensor.solis_inverter_total_energy_charged
- Energy From the Battery (Total): sensor.solis_inverter_total_energy_discharged
- Over Discharge SOC: sensor.solis_inverter_force_discharge_soc
- Force Charge SOC: sensor.solis_inverter_force_charge_soc
